test/tests/functional/pbs_cray_smoketest.py

```diff
@@ -0,0 +1,195 @@
+# trademark licensing policies.
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+import os
+
+
+@tags('cray', 'smoke')
+class TestCraySmokeTest(TestFunctional):
+
+    """
+    Set of tests that qualifies as smoketest for Cray platform
+    """
+
+    def setUp(self):
+        if not self.du.get_platform().startswith('cray'):
+            self.skipTest("Test suite only meant to run on a Cray")
+        TestFunctional.setUp(self)
+
+    @tags('cray', 'smoke')
+    def test_cray_login_job(self):
+        """
+        Check output of apstat -rn on Cray with no queued job.
+        The compute node summary should show that there are no nodes in use,
+        held, or down.  Check the that the number of up nodes is the same as
+        the number of available nodes.
+
+        Check pbsnodes -av output on a Cray system with no queued job.
+        pbsnodes output should show that nodes are free and resources are
+        available.
+
+        Submit a simple sleep job that requests to run on a login node
+        and expect that job to go in running state on a login node.
+        Verify that the job runs to completion and check job output/error.
+        """
+        cu = CrayUtils()
+        self.assertEqual(cu.count_node_in_state('resv'), 0,
+                         "No compute node should be having ALPS reservation")
+        self.assertEqual(cu.count_node_in_state('use'), 0,
+                         "No compute node should be in use")
+        nodes_up_apstat = cu.count_node_in_state('up')
+        nodes = self.server.filter(NODE,
+                                   {ATTR_rescavail + '.vntype':
+                                    'cray_compute'})
+        self.assertEqual(nodes_up_apstat,
+                         len(nodes[ATTR_rescavail + '.vntype=cray_compute']))
+
+        nodes = self.server.status(NODE)
+        for node in nodes:
+            self.assertEqual(node['state'], 'free')
+            self.assertEqual(node['resources_assigned.ncpus'], '0')
+            self.assertEqual(node['resources_assigned.mem'], '0kb')
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+        j1 = Job(TEST_USER, {ATTR_l + '.vntype': 'cray_login',
+                             ATTR_N: 'cray_login'})
+
+        scr = []
+        scr += ['echo Hello World\n']
+        scr += ['/bin/sleep 5\n']
+
+        sub_dir = "/tmp"
+        j1.create_script(scr)
+        jid1 = self.server.submit(j1, submit_dir=sub_dir)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+        # fetch node name where the job is running and check that the
+        # node is a login node
+        self.server.status(JOB, 'exec_vnode', id=jid1)
+        vname = j1.get_vnodes()[0]
+        self.server.expect(NODE, {ATTR_rescavail + '.vntype': 'cray_login'},
+                           id=vname, max_attempts=1)
+
+        cu = CrayUtils()
```

Comment:
I think we should just add a new method to CrayUtils to just run 'apstat -rn' instead of creating another object. That new method can then update CrayUtils internal dictionaries/lists and even return them. What do you think?

Reply:
The second object was a mistake when I moved the test_cray_no_queued_job into test_cray_login_job.  I have made test_cray_no_queued_job as a separate test.